src/pages/LoginAndRegister/GuestAccountRedirect.vue

```diff
@@ -44,14 +48,18 @@ export default {
 					.join('&');
 
 				// Check to see if user is authenticated
-				if (!data?.my?.userAccount?.id) {
+				if (!dataQuery?.my?.userAccount?.id) {
+					const currentRoute = data[1].value ?? data[1] ?? {};
+					const username = currentRoute.query?.username ?? '';
+
 					return Promise.reject({
 						path: '/ui-login',
 						query: {
 							loginHint: `login|${JSON.stringify({
 								guest: true,
 							})}`,
 							doneUrl: `${path}?${queryString}`,
+							username,
```

Comment:
Related to the other comment, maybe this can be passed above in the `loginHint` JSON.

Reply:
Same as previous comment, different purpose... also, this will complicate driving query as `username` will be part of the `loginHint` param....